scripts/deploy/deployEASSchemas.ts

```diff
@@ -0,0 +1,118 @@
+import { execSync } from 'node:child_process';
+
+import {
+  allSchemas,
+  encodeNameSchemaAttestation,
+  NAME_SCHEMA_UID,
+  NULL_EAS_REF_UID,
+  NULL_EVM_ADDRESS
+} from '@charmverse/core/protocol';
+import dotenv from 'dotenv';
+import { task } from 'hardhat/config';
+import type { Address } from 'viem';
+import { createWalletClient, http, isAddress, parseEventLogs, publicActions } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+
+import type { SupportedChains } from '../../lib/connectors';
+import { getConnectorFromHardhatRuntimeEnvironment, getConnectorKey, getEasUrl } from '../../lib/connectors';
+
+dotenv.config();
+
+const PRIVATE_KEY = (
+  process.env.PRIVATE_KEY?.startsWith('0x') ? process.env.PRIVATE_KEY : `0x${process.env.PRIVATE_KEY}`
+) as `0x${string}`;
+
+task('deployEASSchemas', 'Deploys or updates the BuilderNFTSeasonOne contracts').setAction(async (taskArgs, hre) => {
+  const connector = getConnectorFromHardhatRuntimeEnvironment(hre);
+
+  await hre.run('compile');
+
+  const viem = hre.viem;
+
+  const account = privateKeyToAccount(PRIVATE_KEY);
+  const walletClient = createWalletClient({
+    account,
+    chain: connector.chain,
+    transport: http(connector.rpcUrl)
+  }).extend(publicActions);
+
+  console.log('Using account:', account.address, 'on chain:', connector.chain.name);
+
+  if (!isAddress(connector.easAttestationContract as string)) {
+    throw new Error(`No EAS Contract found for chain ${connector.chain.name}:${connector.chain.id}`);
+  }
+
+  const easContract = await viem.getContractAt('IEAS', connector.easAttestationContract as `0x${string}`);
+
+  const easSchemaRegistryAddress = await easContract.read.getSchemaRegistry();
+
+  const easRegistryContract = await viem.getContractAt('ISchemaRegistry', easSchemaRegistryAddress);
+
+  console.log('Deploying the resolver contract...');
+  const deployedResolver = await viem.deployContract(
```

Comment:
Cleaned up deploy script, viem has inbuilt encoding of deploy data and resolution of the ABI

No more need to path.resolve the ABI